tests/src/org.hibernate.orm/hibernate-core/7.0.0.Final/src/test/java/org_hibernate_orm/hibernate_core/OrmAnnotationHelperTest.java

```diff
@@ -0,0 +1,147 @@
+                attributes.put(attributeDescriptor.getName(),
+                    Array.newInstance(typeDescriptor.getValueType().getComponentType(), 0));
+            } else {
+                if (valueType == int.class || valueType == Integer.class) {
+                    attributes.put(attributeDescriptor.getName(), 0);
+                } else if (valueType == long.class || valueType == Long.class) {
+                    attributes.put(attributeDescriptor.getName(), 0L);
+                } else if (valueType == boolean.class || valueType == Boolean.class) {
+                    attributes.put(attributeDescriptor.getName(), false);
+                } else if (valueType == short.class || valueType == Short.class) {
+                    attributes.put(attributeDescriptor.getName(), 0);
+                } else if (valueType == byte.class || valueType == Byte.class) {
+                    attributes.put(attributeDescriptor.getName(), 0);
+                } else if (valueType == char.class || valueType == Character.class) {
+                    attributes.put(attributeDescriptor.getName(), 'c');
+                } else if (valueType == float.class || valueType == Float.class) {
+                    attributes.put(attributeDescriptor.getName(), 0f);
+                } else if (valueType == double.class || valueType == Double.class) {
+                    attributes.put(attributeDescriptor.getName(), 0d);
+                }
+            }
+        }
+        return attributes;
+    }
+
+    private static void createInstances(AnnotationDescriptor<?> annotationDescriptor, AttributeDescriptor<?> attributeDescriptor, Map<String, Object> attributes, BasicModelsContextImpl ctx) {
+
+        for (Annotation usage : usages(ctx, annotationDescriptor, attributes)) {
+            instanceViaUsage(attributeDescriptor, ctx, usage);
+        }
+
+        instanceViaCtors(annotationDescriptor);
+    }
+
+    private static void instanceViaCtors(AnnotationDescriptor<?> annotationDescriptor) {
+        if (annotationDescriptor instanceof MutableAnnotationDescriptor<?, ?> mad) {
+
+            try {
+                Constructor<?> constructor = mad.getMutableAnnotationType().getConstructor(annotationDescriptor.getAnnotationType(), ModelsContext.class);
+                constructor.newInstance(null, null);
+            } catch (Exception e) {
+                // ignore it
+            }
+            try {
+                Constructor<?> constructor = mad.getMutableAnnotationType().getConstructor(ModelsContext.class);
+                constructor.newInstance(null);
+            } catch (Exception e) {
+                // ignore it
+            }
+        }
+    }
+
+    private static List<Annotation> usages(BasicModelsContextImpl ctx, AnnotationDescriptor<?> annotationDescriptor, Map<String, Object> attributes) {
+        List<Annotation> usages = new ArrayList<>();
+        try {
+            usages.add(annotationDescriptor.createUsage(ctx));
```

Comment:
I'm not familiar with the inner workings of hibernate. Can you point me to the definitions of `org.hibernate.models.spi.AnnotationDescriptor.createUsage`?